File: C_competitve_coding/jumping_on_the_clouds.c
//jumping on the clouds- https://www.hackerrank.com/challenges/jumping-on-the-clouds-revisited/problem
#include<stdio.h>
#include<stdlib.h>
int main(){
	int c[8] ={0,0,1,0,0,1,1,0};
	int k ,i , index = 0 , n , e = 100;
	int val= 0;
	printf("\nEnter k");
	scanf("%d",&k);
	n = sizeof(c)/sizeof(c[0]);
	for(i = 0;i<n;i = index){
		index = (i+k)%n; //cloud to be jumped on
		printf("\n--------------------");
		printf("\njumping on index %d", index);

		val = c[index];
		if(val == 1) //thundercloud
			e = e-1-2;
		else 
			e = e-1;
		printf("\n energy is %d ", e);
		if(index == 0)
			break;  //back to the first node

	}
	printf("\nEnergy is %d ", e);
	return 0;
}

File: python_competitve_coding/jumping_on_the_clouds.py
#In python
def jumpingOnClouds(c, k):
    index =0
    i= 0
    e =100
    val =0
    n = len(c)
    
    for i in range(0,n):
        i = index
        index = (i+k)%n #cloud to be jumped on
        val = c[index]
        if(val == 1): #thundercloud
            e = e-1-2
        else:
            e = e-1
        if(index == 0):
            break  #back to the first node
    print(e)
    
    
c = [0,0,1,0,0,1,1,0]
k = int(input())
jumpingOnClouds(c,k)
